Possibility to search instances in public tab

thebigG opened the pull request.

Bnyro commented on src/components/instances_page.rs:
```diff
@@ -90,8 +119,9 @@ impl SimpleComponent for InstancesPage {
     fn update(&mut self, msg: Self::Input, sender: ComponentSender<Self>) {
         match msg {
             InstancesPageInput::FetchInstances => {
+                let filter =  self.instances_search_buffer.text().as_str().to_owned().clone();
```
Please lowercase the string, so that users still get results when using caps lock.

Bnyro replied:
Cloning should not be necessary here.

Bnyro commented on src/api/instances.rs:
```diff
@@ -22,6 +22,7 @@ pub fn fetch_instances() -> std::result::Result<Vec<Instance>, reqwest::Error> {
             .linked
             .iter()
             .filter(|instance| instance.software == Some("lemmy".to_owned()))
+            .filter(|instance| instance.domain.clone().contains(query_filter))
```
please convert the domain name to lowercase.

Bnyro commented on src/components/instances_page.rs:
```diff
@@ -40,13 +41,39 @@ impl SimpleComponent for InstancesPage {
+                    set_vexpand: true,
+                    set_hexpand: true,
+                    gtk::Box {
+
```
nit: please remove that empty line.

Bnyro commented on src/components/instances_page.rs:
```diff
@@ -40,13 +41,39 @@ impl SimpleComponent for InstancesPage {
+                            set_orientation: gtk::Orientation::Vertical,
+                            set_spacing: 5,
+                            set_vexpand: true,
+                        },
+
+                    }
+
+                    }
+
+                    -> {
```
nit: please fix the indentation here and remove the new lines

thebigG commented:
Thanks so much for the feedback @Bnyro :+1: .

I believe I addressed all concerns; hopefully I didn't miss any.

When it comes to formatting, I think we should also integrate fortmat checking into CI. I can probably take a stab at it if you're not working on it. Should be pretty easy since we have a Dockerfile already that captures the build environment.

Bnyro commented:
Yes, I think that would be a good idea.
Unfortunately relm4 doesn't yet have an inbuilt support for being formatted, but the rest of the code can be checked for formatting errors.

Bnyro commented on src/api/instances.rs:
```diff
@@ -22,6 +22,7 @@ pub fn fetch_instances() -> std::result::Result<Vec<Instance>, reqwest::Error> {
             .linked
             .iter()
             .filter(|instance| instance.software == Some("lemmy".to_owned()))
+            .filter(|instance| instance.domain.clone().contains(&query_filter.to_lowercase()))
```
The query is being lowercased on each iteration now. It would be better for the performance if we save it as lowercased to a variable before the we iterate over all instances.

thebigG replied:
Fixed: ee859e01c5ecff2e8f0a60e19f80c56a2caf92cd

Bnyro commented on src/api/instances.rs:
```diff
@@ -22,6 +22,7 @@ pub fn fetch_instances() -> std::result::Result<Vec<Instance>, reqwest::Error> {
             .linked
             .iter()
             .filter(|instance| instance.software == Some("lemmy".to_owned()))
```
I think it would be better if we combine the two filter methods into one, due to performance reasons.

thebigG replied:
Fixed: ee859e01c5ecff2e8f0a60e19f80c56a2caf92cd

Bnyro commented on src/components/instances_page.rs:
```diff
@@ -40,12 +41,32 @@ impl SimpleComponent for InstancesPage {
+                            set_orientation: gtk::Orientation::Vertical,
+                            set_spacing: 5,
+                            set_vexpand: true,
+                            },
+                        }
```
nit: these two lines should be indented one tab less

thebigG replied:
Fixed: bc468b7240848624007b3173fc5b3bd1d99a0b4a

Bnyro commented:
Generally it would be really useful if we had a GitHub action that attempts to build the app and fails if there's something completely wrong, probably the Docker image would be useful too for that purpose.
If you'd like to work on that, that'd be great, otherwise I'll see what I can do in the next few days.

Bnyro commented:
Thank you!
